### DISCUSSION TITLE
Can the commission for the withdrawal of funds from miner balance be taken from the funds themselves?

### DISCUSSION THREAD

**Discussion by Shekelme** [filecoin-project/lotus#8551] [Ideas] [answered]
> s1@s1:~$ lotus-miner actor withdraw
> ERROR: mpool push: failed to push message: not enough funds (required: 0.015015701544215724 FIL, balance: 0.011550844724641968 FIL): not enough funds to execute transaction

Meanwhile there are 6 FIL available on miner balance. Can the commission for the withdrawal of funds from this balance be taken from the funds themselves? BTW why the fee is so big?

**Comment by LexLuthr** [comment, marked as answer]
@Shekelme The fee cannot be taken from the funds as that is locked up. Actor will not release the fund or fee till a message changes the state for it. With current message syntax a from address must have funds to execute the message. So, this might require more efforts than it is worth. As for the fee being high, it could be to avoid delays in executing the message. I will look it up.

**Reply by LexLuthr** [reply]
@Shekelme So, the command uses MpoolPushMessage API which in turn guesses the gas fee. This could be the reason for the high free during that particular time. Idea behind it to make sure message gets executed as soon as possible to avoid delays in withdrawing funds. This could also lead to high gas fee at times.

**Comment by LexLuthr** [comment]
I am archiving this discussion as there has been no new input from the author for more than 7 days.
